src/sse.ts

```diff
@@ -0,0 +1,112 @@
+import { Response } from "express";
+import { z } from "zod";
+import { EmptySchema, FlatObject } from "./common-helpers";
+import { contentTypes } from "./content-type";
+import { Handler } from "./endpoint";
+import { EndpointsFactory } from "./endpoints-factory";
+import { IOSchema } from "./io-schema";
+import { Middleware } from "./middleware";
+import { ResultHandler } from "./result-handler";
+import {
+  ensureHttpError,
+  getPublicErrorMessage,
+  logServerError,
+} from "./result-helpers";
+
+type EventsMap = Record<string, z.ZodTypeAny>;
+
+interface Emitter<E extends EventsMap> extends FlatObject {
+  isClosed: () => boolean;
+  emit: <K extends keyof E>(event: K, data: z.input<E[K]>) => void;
+}
+
+const makeEventSchema = (event: string, data: z.ZodTypeAny) =>
+  z.object({
+    data,
+    event: z.literal(event),
+    id: z.string().optional(),
+    retry: z.number().int().positive().optional(),
+  });
+
+const formatEvent = <E extends EventsMap>(
+  events: E,
+  event: keyof E,
+  data: unknown,
+) =>
+  makeEventSchema(String(event), events[event])
+    .transform((props) =>
+      [
+        `event: ${props.event}`,
+        `data: ${JSON.stringify(props.data)}`,
+        "",
+        "", // empty line: events separator
+      ].join("\n"),
+    )
+    .parse({ event, data });
+
+const headersTimeout = 1e4; // 10s to respond with a status code other than 200
+const ensureStream = (response: Response) =>
+  response.headersSent ||
+  response
+    .type(contentTypes.sse)
+    .setHeader("cache-control", "no-cache")
+    .setHeader("connection", "keep-alive")
+    .flushHeaders();
```

Comment:
this method is not implemented by response mocking library.
Made an issue: https://github.com/eugef/node-mocks-http/issues/312
and a PR for simpler workaround: https://github.com/eugef/node-mocks-http/pull/313